### PR TITLE
[Important] Fix a major bug

### PR DESCRIPTION
**Pull request by kuol**
I was tracking why my R code gave me reasonable answer but Alteryx is not (giving negative answers for the portfolio problems -- which should be values between 0 to 1), then I found this bug. I'm not sure how this will affect LP stuff (beta release), but here's the illustration of the problem:
```
> inputs$O
  variable coefficient lb  ub type
1  Field_2    1.078136  0 Inf    C
2  Field_3    1.092909  0 Inf    C
3  Field_4    1.119773  0 Inf    C
4  Field_5    1.123636  0 Inf    C
5  Field_6    1.121318  0 Inf    C
6  Field_7    1.091773  0 Inf    C
7  Field_8    1.141227  0 Inf    C
8  Field_9    1.128955  0 Inf    C
> AlteryxPrescriptive:::fixMatrixO(inputs$O)
  variable coefficient lb ub type
1  Field_2    1.078136  0  1    C
2  Field_3    1.092909  0  1    C
3  Field_4    1.119773  0  1    C
4  Field_5    1.123636  0  1    C
5  Field_6    1.121318  0  1    C
6  Field_7    1.091773  0  1    C
7  Field_8    1.141227  0  1    C
8  Field_9    1.128955  0  1    C
```
`fixMatrixO` is a function in `processData.R`. It was called very early one when passing the data. 
As you can see here `ub`'s value has changed from `Inf` to `1`. This is due to the old code here:
```
  ub <- rep(Inf, nVar)
  if ('ub' %in% names(O)){
    ub[!is.na(O$ub)] <- O$ub[!is.na(O$ub)]
  }
```
Basically, `O$ub` is interpreted as factor when read into R, since `ub` were implicitly treated as `numeric` due to `rep(Inf, nVar)`, it made `ub == rep(1, nVar)`. 

After the fix, the solution looks reasonable now (between 0, 1), however, I haven't figured out why `ub=1` screws things up. For portfolio optimization problem, I'm solving the weights (percentage of my money that should be invested in certain asset)  and I'm having constraints that all weights added up equal to 1. So I would think `ub=Inf` and `ub=1` are both fine. 

I haven't fully tested any LP problem, but it is true that we didn't really test and make sure the R results and Alteryx results actually match.... I'm hoping this won't affect beta.

### CODE REVIEW

**Comment by ramnathv** [comment]
Good catch Kuo. We had fixed a bunch of factor issues earlier. But this is a subtle one, since we really don't enter the upper bound when it equals the default of `Inf`. I wouldn't say this is a critical bug. So we can roll with the current code for the beta. However, this underlines the need to write more tests around `inputs` getting transformed correctly, before being handed off to the `constructModel` function.

I will close this PR for now, as I have to merge the `feature/reporting` branch to `master` first.  I will make the fix right after that.

**Comment by kuol** [comment]
Thanks Ramnath. Any ideas why `ub==1` vs `ub==Inf` give completely different results? Since we require x's >=0 and sum(x) = 1, no x can be greater than 1 anyway.

**Comment by kuol** [comment]
Also, we currently specifies `lb`, `ub`, `type` explicitly (and lot of them are `ub = (Inf, Inf ...)` ) in the sample workflows/tests we handed out to other people. Are we gonna change the sample now?
